mcs/class/referencesource/mscorlib/system/activator.cs

```diff
@@ -74,7 +74,7 @@ private Activator()
                 throw new ArgumentNullException("type");
             Contract.EndContractBlock();
 #if !FULL_AOT_RUNTIME
-            if (type is System.Reflection.Emit.TypeBuilder)
+            if (RuntimeFeature.IsDynamicCodeSupported && type is System.Reflection.Emit.TypeBuilder)
```

Comment:
We could also fix the linker not to mark for `is` (but that can be done later)

Reply:
That would not work, it's not that trivial ;-)

The only way to make this work _inside_ an active method is to use dead-code elimination - including variable elimination, which again means that you need to adjust all `ldloc` / `stloc` instructions in the method.   That's not as scary as it may sound - and my new module already does this.

To use the existing linker with some minor changes (like for instance my abandoned draft https://github.com/mono/linker/pull/555, with `return false` added) is to move the `is` into a separate helper method and then make the linker rewrite that.

So for instance something like

```
static bool IsTypeBuilder (Type type) => type is TypeBuilder;
```

and then some XML description to tell the linker to turn that into `return false`:

```
<method name="IsTypeBuilder" action="return-false" />
```

Reply:
Btw. I spent quite some time researching exactly this `x is y` situation and the tricky part is to distinguish between the trivial cases and those which require flow analysis and dead-code elimination.

This particular example here is simple enough that it could be rewritten - but detecting that is not easy.

In IL, the `isinst Foo` leaves an instance of `Foo` on the stack (but we're going to remove `Foo`, so it's a non-existing type).  The trivial case is something like this (`static bool IsWeakInstanceOf<T> (object obj) => obj is T`):

```
        IL_0000:  ldarg.0 
        IL_0001:  isinst !!0
        IL_0006:  ldnull 
        IL_0007:  cgt.un 
        IL_0009:  ret 
```

But if the `isinst` is followed by a `stloc`, `call` or `ret`, then we might be in trouble.

During my research, I've seen some cases in complex corlib methods where all my detection logic failed, so I only used it in a few cases that we explicitly whitelisted.

Reply:
It's actually quite trivial, we just don't mark types as instantiated when used from is instruction. That way they will be marked but with no dependencies

Reply:
Yeah mean keeping the `TypeBuilder` type, but not it's dependencies?

Reply:
yeah, not even the ctors or fields (linker has this infrastructure in place already)